README
Pod::Tree - Tree-based POD parser

DESCRIPTION

Modules

    Pod::Tree parses a POD into a static syntax tree.
    Applications walk the tree to recover the structure and content of
    the POD. 

    Pod::Tree::HTML walks a Pod::Tree and translates it to HTML.

    Pod::Tree::Pod walks a Pod::Tree and translates it back to POD.

    Pod::Tree::Perl* translate the PODs in the Perl distribution to HTML.

Executables

    pod2html translates a POD file to an HTML file.

    pods2html walks a directory tree, 
    translates all the PODs that it finds into HTML files,
    and puts the HTML files into a parallel directory tree.

    perl2html translates the PODs in the Perl distribution to HTML


PREREQUISITES
Perl 5
File::Find
HTML::Stream
IO::File
IO::String
Pod::Escapes
Text::Template


HOW TO BUILD AND INSTALL

    perl Makefile.PL
    make
    make test
    make install


You may see a message like

    Warning: prerequisite HTML::Stream 1.49 not found.

You can ignore this, provided that you actually have HTML::Stream 1.49
or later installed.


TODO
See the ToDo and ToDo.Not files.
Send suggestions, bugs, etc. to swmcd@world.std.com,
or open a ticket at

    http://rt.cpan.org/Dist/Display.html?Queue=Pod-Tree


ACKNOWLEDGMENTS

                        <crazyinsomniac@yahoo.com>
			<joenio@cpan.org>
Paul Bettinger		<paul@n8geil.de>
Sean M. Burke		<sburke@spinn.net>
Brad Choate		<brad@bradchoate.com>
Havard Eidnes		<he@NetBSD.org>
Rudi Farkas		<rudif@bluemail.ch>
Paul Gibeault		<pagibeault@micron.com>
Jay Hannah		<jhannah@omnihotels.com>
Martin Hans		<mah@oticon.dk>
Paul Hawkins		<phawkins@datajunction.com>
Jost Krieger		<Jost.Krieger@ruhr-uni-bochum.de>
Marc A. Lehmann		<pcg@goof.com> 
Jonas Liljegren		<jonas@jonas.rit.se>
Thomas Linden		<tom@co.daemon.de>
Johan Lindstrom		<johanl@bahnhof.se>
Terry Luedtke           <terry_luedtke@nlm.nih.gov>
Rob Napier		<rnapier@employees.org>
Kate L Pugh             <kake@earth.li>
Christopher Shalah	<trance@drizzle.com>
Blair Sutton		<bsdz@numeninest.com>
Johan Vromans		<JVromans@Squirrel.nl>


COPYRIGHT
Copyright (c) 1999-2009 by Steven McDougall. This module is free
software; you can redistribute it and/or modify it under the same
terms as Perl itself.


Changes
Revision history for Perl extension Pod::Tree

1.16    2009 Apr 10
	Pod::Tree::HTML
	- fix to work with IO::File 1.13 (RT 34755, 38307)

	mod2html
	- added to distribution

	Makefile.PL
	- added all required modules to PREREQ_PM


1.15    2007 Jun 24
	Rename pod2html -> podtree2html (RT 18742)

	Pod::Tree::HTML
	- Changed destination resolution to accept a scalar reference,
	  or anything with a 'print' method. (RT 12335)

        podtree2html, pods2html
	- documented HTML elements for CSS rules (RT 6118)

	pod.css
	- add example style sheet


1.14	2007 Jun 12
	Pod::Tree::HTML
	- remove use vars qw(&isa); (RT 14530)
	- document translate() error conditions (RT 12336)

	Makefile.PL
	- add Text::Template prerequisite (RT 26440)


1.13    2007 Apr 06
	Pod::Tree
	- rename the Stream package

	Pod::Tree::HTML
	- change the X<> markup to render as empty, per the perlpod manpage
	- rename the BitBucket package
	- add template support

	Pod::Tree:PerlUtil
	- handle malformed NAME paragraphs more gracefully

	pod2html
	- add template support

	pods2html
	- add template support


1.12	2006 Apr 22
	Pod::Tree::PerlBin
	- ignore non-existant directories on $PATH
	- protect File::Find::find from empty directory list

	Pod::Tree:PerlFunc
	- find the pod/ directory under either perl/ or perl/lib/ (Win32 issue)

	Pod::Tree::PerlLib
	- increase default col_width to 30

	Pod::Tree:PerlTop
	- find the pod/ directory under either perl/ or perl/lib/ (Win32 issue)

	Pod::Tree:PerlUtil
	- clean up whitespace in module names and descriptions

	pods2html
	- recursion check compares absolute paths instead of inodes (Win32 issue)

	t/pods2html.t
	- use rmtree instead of system "rm -r" (Win32 issue)


1.11	2004 Oct
	Pod::Tree
	- recognize blank lines per perlpodspec
	- added support for =head3 and =head4 commands
	- recognize \n, \r, \r\n line endings
	- separate paragraphs on blank (/^\s*$/) lines

	Pod::Tree::HTML
	- added support for =head3 and =head4 commands
	- expose the emit_toc and emit_body methods
	- drop the </li> tags in the TOC.
	  It was the easiest way to fix the nesting problem.
	- added { empty => 1 } option

	Pod::Tree::Node
	- added support for =head3 and =head4 commands
	- consolidate consecutive verbatim paragraphs

	pods2html
	- added --module option
	- get depth from module name, not file path
	- added --empty option

	t/pod2html.t
	t/pods2html.t
	- invoke $Config{perlpath}, not 'perl'

	GENERAL
	- Set default BGCOLOR to ffffff

	- Removed all the
	    use 5.x;
	    require 5.x
	  statements

	- reviewed all the copyright notices

1.10	2003 Jan
	Pod::Tree
	- documentation typos
	- dropped the require 5.6.0; statement
	
	Pod::Tree::HTML
	- added support for $link_map->url methods
	- deprecated        $link_map->map methods
	- exposed escape_2396 and assemble_url methods
	- documentation

	Pod::Tree::Node
	- documentation typos

	Pod::Tree::Pod
	- dropped the require 5.6.0; statement

	t/mapper.t
	t/mapper.d
	- created

	
1.09	2002 Jul
	Makefile.PL
	- traded in ABSTRACT for ABSTRACT_FROM

	Pod::Tree::Node
	- recoded _unescape_text to use Pod::Escapes
	- expand the HTTP domain to include e.g. mailto:

	Pod::Tree::Pod
	- changed _escape to handle non-ASCII chars

	Pod::Tree::PerlUtils
	- Handle module names that contain dashes

	Pod::Tree::HTML
	- Handle PODs that have "name--description" with multiple dashes
	- Change default bgcolor to white
	- added css => $url option

	perl2html
	- traded in hard-coded /:/ for /$Config{path_sep}/

	pod2html
	- added -css URL option

	pods2html
	- added -css URL option
	- fixed recursion check on Win32

	t/pod2html.d
	t/pod2html.t
	- created

	t/pods2html.t
	- enable some tests on Win32
	- wrote RDiff() to ignore CVS directories


1.08	2001 May 30
	t/pods2html.t
	- skip test on Win32


1.07	2001 May 29

	Makefile.PL
	- added EXE_FILES key
	- added PREREQ_PM key

	Pod::Tree
	- preserve paragraph endings
	- preserve non-POD content
	- added has_pod() method
	
	Pod::Tree::Node
	- added code() and is_code()
	- added get_brackets() method
	- added get_raw() method
	- added clone() method
	- added set_filename()/get_filename() methods
	- added get_domain() and support for L<http://...>
	- carry =pod and =cut in command nodes
	- SplitBar() split(/\|/,,2) to protect links with a verbar
	
	Pod::Tree:HTML
	- fixed load_paragraphs() call in resolve_source()
	- added depth option
	- render  =for image foo.gif  as  <img src="foo.gif">
	- rewhacked _emit_link to support target domains

	Pod::Tree::Pod
	- correctly reconstructs L<> markups

	Pod::Tree::PerlDist
	- added perl* links for the README.* files

	Pod::Tree::PerlFunc
	- fixed ShiftItems() to handle the getpw* items in perlfunc.pod

	Pod::Tree::PerlTop
	- added _get_dist_links() to pick up the README.* files

	pods2html
	- fixed relative links
	- don't translate foo.pm if foo.pod exists
	- works if HTMLDir is a subdir of PodDir
	- create HTMLDir with File::Path::mkpath

	t/
	- moved all the test files into .d/ directories


1.06	2000 Oct 17

	Added Pod::Tree::Pod

	Pod::Tree::Node
	- fixed SpitBar()


1.05	2000 Oct 12

	Pod::Tree::Node
	- Added $VERSION to quiet CPAN.pm


1.04	2000 Oct 11
	
	Pod::Tree::HTML
	- Added $VERSION to quiet CPAN.pm


1.03	2000 Oct 09

	- Cleaned up $VERSION variables (we hope)

	Pod::Tree::HTML
	- fixed \$pod bug in new()

	Pod::Tree::PerlBin
	- fixed xsubpp bug

	Pod::Tree::PerlFunc
	- fixed to handle do() properly

	Pod::Tree::PerlPod
	- compress whitespace in description lines
	
	pods2html
	- Fixed to handle relative paths

	perl2html
	- Fixed to handle relative paths


1.02	2000 May 19

	Pod::Tree::Node
	- added support for x<<>> style markups

	Pod::Tree::PerlFunc
	- added <dl></dl> tags to lists in perlfunc.pod sub-pages
	- sort function names within alphebetical index paragraphs

	Pod::Tree::PerlLib
	- scan for .pod files under @INC
	- added stop_file list

	Pod::Tree::PerlPod
	- don't convert PODs under /ext/


1.01	2000 Mar 11

	Pod::Tree
	- added limit => n option

Makefile.PL
use ExtUtils::MakeMaker;

# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

WriteMakefile(NAME	      => 'Pod::Tree',
	      VERSION_FROM    => 'lib/Pod/Tree.pm', # finds $VERSION
	      DISTNAME        => 'Pod-Tree',
	      ($] >= 5.005 ?
	       (ABSTRACT_FROM => 'lib/Pod/Tree.pm',
		AUTHOR        => 'Steven McDougall (swmcd@world.std.com)') : 
	       ()),
	      clean           => { FILES => "t/pod2html.d/*.html* t/pods2html.d/A t/pods2html.d/html_act* t/pods2html.d/podR/HTML t/*/*.act lib/Pod/Tree/*~" },
	      dist            => { COMPRESS => 'gzip --best',
			 	   SUFFIX   => '.gz' },
	      EXE_FILES	      => [qw(podtree2html pods2html perl2html)],
	      PREREQ_PM       => { HTML::Stream   => 1.49,
	                           Pod::Escapes   => 1.02,
				   Text::Template => 1      },
);
